src/Perlang.Tests.Integration/EvalHelper.cs

```diff
@@ -303,32 +303,34 @@ internal static object EvalWithArguments(string source, params string[] argument
 
         private static void AssertFailScanErrorHandler(ScanError scanError)
         {
-            throw new Exception("ScanError occurred. See inner exception for details.", scanError);
+            // Note: does not contain stack trace, so no point in wrapping in Exception() at this point.
+            throw scanError;
         }
 
         private static void AssertFailParseErrorHandler(ParseError parseError)
         {
-            throw new Exception("ParseError occurred. See inner exception for details.", parseError);
+            // Note: does not contain stack trace, so no point in wrapping in Exception() at this point.
+            throw parseError;
         }
 
         private static void AssertFailNameResolutionErrorHandler(NameResolutionError nameResolutionError)
         {
-            throw new Exception("NameResolutionError occurred. See inner exception for details.", nameResolutionError);
+            throw new Exception($"NameResolutionError occurred: {nameResolutionError.Message}", nameResolutionError);
         }
 
         private static void AssertFailRuntimeErrorHandler(RuntimeError runtimeError)
         {
-            throw new Exception("RuntimeError occurred. See inner exception for details.", runtimeError);
+            throw new Exception($"RuntimeError occurred: {runtimeError.Message}", runtimeError);
         }
 
         private static void AssertFailValidationErrorHandler(ValidationError validationError)
         {
-            throw new Exception("ValidationError occurred. See inner exception for details.", validationError);
+            throw new Exception($"ValidationError occurred: {validationError.Message}", validationError);
         }
 
         private static bool AssertFailCompilerWarningHandler(CompilerWarning compilerWarning)
         {
-            throw new Exception("CompilerWarning occurred. See inner exception for details.", compilerWarning);
+            throw new Exception($"CompilerWarning occurred: {compilerWarning.Message}", compilerWarning);
```

Comment:
These hunks makes the inner exception message be immediately apparent in e.g. Rider's "last test result" overlay when editing code, which is nice since it makes it super-easy to see why a particular test is failing while you are working on it:

![image](https://user-images.githubusercontent.com/630613/150668647-cb6b9364-025d-4f0a-b527-d17af9321e63.png)